lib/core/include/termiosUtil.hpp

```diff
@@ -0,0 +1,99 @@
+#ifndef _TERMIOS_UTIL_HPP_
+#define _TERMIOS_UTIL_HPP_
+
+#include <unistd.h>
+#include <errno.h>
+#include <termios.h>
+
+#ifdef __cplusplus
+
+namespace irods
+{
+    // class termiosUtil - at this time, this class encapsulates the work
+    // which has to be done in order to turn echo on or off for a specifc
+    // file desctiptor.  The call can be expanded to encapsulate other
+    // functionality of termios(3).
+
+    class termiosUtil {
+        public:
+            // Class has to be instantiated with a file descriptor.
+            // The default constructor is invalid.
+            explicit termiosUtil(int fd) :
+                valid_(false),
+                errno_copy_(0),
+                fd_(fd),
+                save_c_lflag_(0)
+            {
+                memset(&tty_, 0, sizeof(tty_));
+                if (isatty(fd_))
+                {
+                    if (tcgetattr( fd_, &tty_ ) == 0)
+                    {
+                        save_c_lflag_ = tty_.c_lflag;
+                        valid_ = true;
+                    }
+                    else
+                    {
+                        errno_copy_ = errno;
+                        valid_ = false;
+                    }
+                }
+                else
+                {
+                    errno_copy_ = errno;
+                    valid_ = false;
+                }
+            };
+
+        private:
+            // Invalid constructor.  Ensures instantiation with fd parameter
+            termiosUtil(void) {};
+
+        public:
+            virtual ~termiosUtil(void) = default;
+            int getError(void) const     { return errno_copy_; };
+            bool getValid(void) const     { return valid_; };
+
+            bool echoOff(void)
```

Comment:
`void` is probably not needed.  An empty set of parenthesis is pretty clear.

Reply:
Again - this is a choice.  It stays.